src/core/view/Mask.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Xournal++
+ *
+ * Virtual class for showing overlays (e.g. active tools, selections and so on)
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#include <cairo.h>
+
+#include "util/Range.h"
+#include "util/raii/CairoWrappers.h"
+
+namespace xoj::view {
+
+class Mask {
```

Comment:
Needs doc comments.

Thanks for adding this, we could probably simplify several of the tool handlers with this.

In a follow-up: it would also be useful to have the ability to resize the cairo_surface (such as is done in VerticalToolHandler).

Reply:
> In a follow-up: it would also be useful to have the ability to resize the cairo_surface (such as is done in VerticalToolHandler).

Yes I agree. Also in the other tools handlers actually: the tool handlers should listen to zoom or scroll events and adapt their masks accordingly, like the VerticalToolHandler does.

Reply:
Also: This class could be placed in util rather than core/view. Not sure which makes more sense.